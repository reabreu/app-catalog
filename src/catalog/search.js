import React from "react";
import { SearchInput } from "./styles";
import debounce from "lodash.debounce";

const SEARCH_DEBOUNCE_INTERVAL = 300;

export const Search = ({ searchTerm, onChangeHandler }) => {
  const debouncedDispatch = debounce(onChangeHandler, SEARCH_DEBOUNCE_INTERVAL);

  return (
    <SearchInput
      onChange={(e) => debouncedDispatch(e.target.value)}
      defaultValue={searchTerm}
      placeholder={"Search"}
    />
  );
};
