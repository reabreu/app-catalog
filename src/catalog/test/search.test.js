import React from "react";
import { render, fireEvent, wait, cleanup } from "@testing-library/react";
import { Search } from "../search";

afterEach(cleanup);

describe("<Search /> spec", () => {
  it("Contains search input and sets initial value", () => {
    const { asFragment } = render(
      <Search searchTerm={"foo"} onChangeHandler={() => {}} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("calls debounced onChangeHandler on input change ", async () => {
    const onChangeHandler = jest.fn();

    const { getByRole } = render(
      <Search searchTerm={"foo"} onChangeHandler={onChangeHandler} />
    );

    const input = getByRole("textbox");

    fireEvent.change(input, { target: { value: "a" } });

    await wait(() => expect(onChangeHandler).toHaveBeenCalledWith("a"), {
      timeout: 300,
    });
  });
});
