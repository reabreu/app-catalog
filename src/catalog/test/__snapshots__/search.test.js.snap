// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> spec Contains search input and sets initial value 1`] = `
<DocumentFragment>
  <input
    class="sc-fzoLsD jOfBdy"
    placeholder="Search"
    value="foo"
  />
</DocumentFragment>
`;
